### PR TITLE
Feat risk management

### PR DESCRIPTION
**Pull request by laszlokr**

### CODE REVIEW

File: mgmt_asset/models/asset.py [hunk 1]
```diff
@@ -10,3 +10,11 @@ class MgmtAsset(models.Model):
     _description = "Mgmt Asset"
 
     name = fields.Char()
+    model_name = fields.Selection([
```
Comment (janikvonrotz) [inline mgmt_asset/models/asset.py]: Update the list.

File: mgmt_asset/models/asset.py [hunk 2]
```diff
@@ -10,3 +10,11 @@ class MgmtAsset(models.Model):
     _description = "Mgmt Asset"
 
     name = fields.Char()
```
Comment (janikvonrotz) [inline mgmt_asset/models/asset.py]: This is computed. Use displayname of linked record.

File: mgmt_audit/models/nonconformity.py [hunk 3]
```diff
@@ -9,4 +9,9 @@ class MgmtNonconformity(models.Model):
     _name = "mgmt.nonconformity"
     _description = "Mgmt Nonconformity"
 
-    name = fields.Char()
+    name = fields.Char(required=True)
+    type = fields.Selection([
+        ('major', 'Major'),
+        ('minor', 'Minor'),
+    ], required=True)
+    task_ids = fields.One2many('project.task')
```
Comment (janikvonrotz) [inline mgmt_audit/models/nonconformity.py]: This should be in the `mgmt_audit_project` module.

File: mgmt_audit_project/models/recommendation.py [hunk 4]
```diff
@@ -7,3 +7,6 @@
 
 class MgmtRecommendation(models.Model):
     _inherit = "mgmt.recommendation"
+    _description = "Mgmt Reccomendation"
```
Comment (janikvonrotz) [inline mgmt_audit_project/models/recommendation.py]: Typo.

File: mgmt_risk/models/hazard.py [hunk 5]
```diff
@@ -9,4 +9,6 @@ class MgmtHazard(models.Model):
     _name = "mgmt.hazard"
     _description = "Mgmt Hazard"
 
-    name = fields.Char()
+    name = fields.Char(required=True)
+    severity_id = fields.Many2one('mgmt.severity')
```
Comment (janikvonrotz) [inline mgmt_risk/models/hazard.py]: Move to `mgmt_hazard_risk` module.

File: mgmt_risk/models/risk.py [hunk 6]
```diff
@@ -10,3 +10,19 @@ class MgmtRisk(models.Model):
...
+    description = fields.Text()
+
+    risk_owner_id = fields.Many2one('res.users', required=True)
+    asset_ids = fields.Many2many('mgmt.asset', required=True)
+    severity_id = fields.Many2one('mgmt.severity', required=True)
+    hazard_ids = fields.Many2many('mgmt.hazard', required=True)
+    revision_risk_ids = fields.One2many('mgmt.risk', 'head_risk_id')
+    head_risk_id = fields.Many2one('mgmt.risk')
+    probability_id = fields.Many2one('mgmt.probability', required=True)
+    risk_score = fields.Float(compute='_compute_risk_score', store=True)
```
Comment (janikvonrotz) [inline mgmt_risk/models/risk.py]: Implement the copute method. Based on the formula it calculates the risk score.

File: mgmt_audit/models/statement.py [hunk 7]
```diff
@@ -9,4 +9,10 @@ class MgmtStatement(models.Model):
     _name = "mgmt.statement"
     _description = "Mgmt Statement"
 
-    name = fields.Char()
+    name = fields.Char(required=True)
+    risk_id = fields.Many2one('mgmt.risk')
+    audit_id = field.Many2one('mgmt.audit')
+    attachment_id = fields.Many2one('ir.attachment')
+    requirement_id = fields.Many2one()
```
Comment (janikvonrotz) [inline mgmt_audit/models/statement.py]: Missing model.

File: mgmt_risk/models/risk.py [hunk 8]
```diff
@@ -10,3 +10,19 @@ class MgmtRisk(models.Model):
...
+    color = fields.Integer(compute='_compute_color', store=True)
+    stage = fields.Selection([
+        ('identify', 'Identify'),
+        ('evaluate_threat', 'Evaluate Threat'),
```
Comment (janikvonrotz) [inline mgmt_risk/models/risk.py]: `evaluate`

File: mgmt_risk/models/risk.py [hunk 9]
```diff
@@ -10,3 +10,19 @@ class MgmtRisk(models.Model):
...
+        ('monitor', 'Monitor')
```
Comment (janikvonrotz) [inline mgmt_risk/models/risk.py]: `migitate`

**Review by janikvonrotz — CHANGES_REQUESTED** [review]

**Review by janikvonrotz — COMMENTED** [review]
Rename `mgmt_risk/models/res_config_settings.py` to `res_company`.

Example for `mgmt_risk_formula`:

```python
class ResCompany(models.Model):
    _inherit = 'res.company'

    avalara_api_id = fields.Char(string='Avalara API ID', groups='base.group_system')
    avalara_api_key = fields.Char(string='Avalara API KEY', groups='base.group_system')
    
class ResConfigSettings(models.TransientModel):
    _inherit = 'res.config.settings'

    avalara_api_id = fields.Char(
        related='company_id.avalara_api_id',
        readonly=False,
        string='Avalara API ID',
    )
    avalara_api_key = fields.Char(
        related='company_id.avalara_api_key',
        readonly=False,
        string='Avalara API KEY',
    )
```

**Comment by janikvonrotz** [comment]
And then the same for `mgmt_audit_project_id` in `mgmt_audit_project`

**Comment by laszlokr** [comment]
@janikvonrotz where should I put these top level menu entries?

```
<!-- Audit Menu -->
<menuitem id="menu_management_audit" name="Audit" parent="menu_mgmtsystem_root" sequence="10"/>

<!-- Evaluation Menu -->
<menuitem id="menu_management_evaluation" name="Evaluation" parent="menu_mgmtsystem_root" sequence="20"/>

<!-- Inventory Menu -->
<menuitem id="menu_management_inventory" name="Inventory" parent="menu_mgmtsystem_root" sequence="30"/>

<!-- Reporting Menu -->
<menuitem id="menu_management_reporting" name="Reporting" parent="menu_mgmtsystem_root" sequence="40"/>

<!-- Configuration Menu -->
<menuitem id="menu_management_configuration" name="Configuration" parent="menu_mgmtsystem_root" sequence="50"/>
<menuitem id="menu_configuration_settings" name="Settings" parent="menu_management_configuration" sequence="10"/>
<menuitem id="menu_configuration_systems" name="Systems" parent="menu_management_configuration" action="action_mgmt_system_view" sequence="20"/>
<menuitem id="menu_configuration_audit" name="Audit" parent="menu_management_configuration" sequence="30"/>
<menuitem id="menu_configuration_risk" name="Risk" parent="menu_management_configuration" sequence="40"/>

```

File: mgmt_base/views/menu.xml [hunk 10]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+        <!-- Audit Menu -->
+        <menuitem id="menu_management_audit" name="Audit" parent="menu_mgmtsystem_root" sequence="10"/>
```
Comment (janikvonrotz) [inline mgmt_base/views/menu.xml]: prefix parent with module name.

File: mgmt_base/security/ir.model.access.csv
Comment (janikvonrotz) [inline mgmt_base/security/ir.model.access.csv:1]: Remove this.

File: mgmt_base/README.rst [hunk 11]
```diff
@@ -0,0 +1,26 @@
+.. image:: https://img.shields.io/badge/licence-GPL--3-blue.svg
+    :target: http://www.gnu.org/licenses/gpl-3.0-standalone.html
+    :alt: License: GPL-3
+
+==========
+Mgmt Audit
+==========
+
+Audit your company.
```
Comment (janikvonrotz) [inline mgmt_base/README.rst]: Update readme.

File: mgmt_base/__manifest__.py [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+        Base for the management modules.
+    """,
+    "author": "Mint System GmbH, Odoo Community Association (OCA)",
+    "website": "https://github.com/OCA/management-system",
+    "category": "Management",
+    "version": "17.0.1.0.0",
+    "license": "AGPL-3",
+    "depends": ["mgmtsystem"],
+    "data": [
+        "security/ir.model.access.csv",
```
Comment (janikvonrotz) [inline mgmt_base/__manifest__.py]: Reoe.

**Review by janikvonrotz — CHANGES_REQUESTED** [review]

File: mgmt_risk/__manifest__.py [hunk 13]
```diff
@@ -8,7 +8,7 @@
     "category": "Management",
     "version": "17.0.1.0.0",
     "license": "AGPL-3",
-    "depends": ["mgmtsystem", "hr"],
+    "depends": ["mgmt_base", "hr", "mgmt_asset"],
```
Comment (janikvonrotz) [inline mgmt_risk/__manifest__.py]: According to the spec this is wrong.

File: mgmt_risk/demo/demo.xml [hunk 14]
```diff
@@ -1,19 +1,34 @@
...
-    <record id="demo_hazard0" model="mgmt.hazard">
-        <field name="name">Hazard 0</field>
+    <!-- Create a demo user for risk_owner_id (assuming a user is needed) -->
+    <record id="base_demo_user" model="res.users">
+        <field name="name">Risk Owner User</field>
+        <field name="login">risk_owner_user</field>
+        <field name="groups_id" eval="[(6, 0, [ref('base.group_user')])]"/>
     </record>
 
+    <!-- Create Severity and Probability records first -->
```
Comment (janikvonrotz) [inline mgmt_risk/demo/demo.xml]: Please remove any comment.

File: mgmt_risk/demo/demo.xml [hunk 15]
```diff
@@ -1,19 +1,34 @@
 <odoo>
 
-    <record id="demo_risk0" model="mgmt.risk">
-        <field name="name">Risk 0</field>
-    </record>
-
-    <record id="demo_hazard0" model="mgmt.hazard">
-        <field name="name">Hazard 0</field>
+    <!-- Create a demo user for risk_owner_id (assuming a user is needed) -->
+    <record id="base_demo_user" model="res.users">
```
Comment (janikvonrotz) [inline mgmt_risk/demo/demo.xml]: Please use existing demo user.

File: mgmt_risk/models/res_company.py [hunk 16]
```diff
@@ -0,0 +1,24 @@
...
+
+_logger = logging.getLogger(__name__)
+
+
+class ResCompany(models.Model):
+    _inherit = "res.company"
+
+    mgmt_risk_formula = fields.Selection(
+        [
+            ("multiply", "Severity * Probability"),
```
Comment (janikvonrotz) [inline mgmt_risk/models/res_company.py]: Replace `*` with `x`

File: mgmt_risk/models/risk.py [hunk 17]
```diff
@@ -9,4 +9,30 @@ class MgmtRisk(models.Model):
...
+        ('mitigate', 'Mitigate')
+    ], required=True)
+
+
+    @api.depends('severity_id', 'probability_id')
+    def _compute_risk_score(self):
+        for record in self:
+            severity = record.severity_id.value if record.severity_id else 0
+            probability = record.probability_id.value if record.probability_id else 0
+            if severity + probability > 0:
```
Comment (janikvonrotz) [inline mgmt_risk/models/risk.py]: Please use formula via company record.

**Review by janikvonrotz — CHANGES_REQUESTED** [review]

File: mgmt_risk/models/risk.py (hunk 17 above)
Comment (laszlokr) [reply mgmt_risk/models/risk.py]: @janikvonrotz I was not sure if the formula shown in ERD implies division between multiplication and sum, but considering the `mgmt_risk_formula` field is a selection between multiplication and sum, I altered the method to reflect that and removed the division. Please let me know if this is correct

File: mgmt_risk/models/risk.py [hunk 18]
```diff
@@ -9,4 +9,40 @@ class MgmtRisk(models.Model):
...
+    risk_score = fields.Float(compute='_compute_risk_score', store=True)
+    color = fields.Integer(compute='_compute_color', store=True)
+    stage = fields.Selection([
+        ('identify', 'Identify'),
+        ('evaluate', 'Evaluate'),
+        ('mitigate', 'Mitigate')
+    ], required=True)
+
+
+    @api.depends('severity_id', 'probability_id', 'company_id')
```
Comment (janikvonrotz) [inline mgmt_risk/models/risk.py]: `company_id` should not be in this list.

File: mgmt_risk/models/risk.py [hunk 19]
```diff
@@ -9,4 +9,40 @@ class MgmtRisk(models.Model):
...
+    def _compute_risk_score(self):
+        for record in self:
+            severity = record.severity_id.value if record.severity_id else 0
+            probability = record.probability_id.value if record.probability_id else 0
+
+            company = record.company_id or self.env.company
```
Comment (janikvonrotz) [inline mgmt_risk/models/risk.py]: Make this a oneliner `self.env.company.mgmt_risk_formula`.

File: mgmt_risk/models/risk.py [hunk 20]
```diff
@@ -9,4 +9,40 @@ class MgmtRisk(models.Model):
     _name = "mgmt.risk"
     _description = "Mgmt Risk"
 
-    name = fields.Char()
+    company_id = fields.Many2one('res.company', string='Company', required=True, default=lambda self: self.env.company)
```
Comment (janikvonrotz) [inline mgmt_risk/models/risk.py]: Remove this field. It is not defined in the ERD.

File: mgmt_risk/models/risk.py [hunk 21]
```diff
@@ -9,4 +9,40 @@ class MgmtRisk(models.Model):
...
+    @api.depends('severity_id', 'probability_id', 'company_id')
+    def _compute_risk_score(self):
+        for record in self:
+            severity = record.severity_id.value if record.severity_id else 0
+            probability = record.probability_id.value if record.probability_id else 0
+
+            company = record.company_id or self.env.company
+            formula = company.mgmt_risk_formula
+
+            if severity > 0 and probability > 0:
```
Comment (janikvonrotz) [inline mgmt_risk/models/risk.py]: This check is unnecessary.

File: mgmt_risk/demo/demo.xml [hunk 22]
```diff
@@ -1,19 +1,137 @@
...
 
-    <record id="demo_hazard0" model="mgmt.hazard">
-        <field name="name">Hazard 0</field>
+    <record id="demo_inventory_category_servers" model="maintenance.equipment.category">
+        <field name="name">Servers</field>
     </record>
 
-    <record id="demo_severity0" model="mgmt.severity">
-        <field name="name">Severity 0</field>
+    <record id="demo_severity5" model="mgmt.severity">
```
Comment (janikvonrotz) [inline mgmt_risk/demo/demo.xml]: You need to define all probabilities and severities from 1 to 5.

File: mgmt_risk/demo/demo.xml
Comment (janikvonrotz) [inline mgmt_risk/demo/demo.xml:1]: The demo data needs to be splitted into the modules.

**Review by janikvonrotz — CHANGES_REQUESTED** [review]

**Comment by laszlokr** [comment]
@janikvonrotz I have a problem defining demo data for mgmt_asset because of the required field `equipment_assign_to`

In the current version I get this error:
```
ValueError: Invalid field 'equipment_assign_to' on model 'maintenance.equipment'
```
If it's not addded, then the error is: 
```
psycopg2.errors.NotNullViolation: null value in column "equipment_assign_to" violates not-null constraint
```

For the same reason the `mgmt_risk` demo data also fails to load because it is referencing this inventory category, so for now I commented it

**Comment by laszlokr** [comment]
@janikvonrotz also for the views, it was not specified where the kanban view should be added and where not. If you could, please clarify

**Comment by janikvonrotz** [comment]
@laszlokr The specification has the details on the kanban view.

For the demo data of maintenance I guess you need provide a value for the field.

File: mgmt_asset/demo/demo.xml [hunk 23]
```diff
@@ -1,7 +1,14 @@
...
-    <record id="demo_asset0" model="mgmt.asset">
-        <field name="name">Asset 0</field>
+    <record id="demo_inventory_category_servers" model="maintenance.equipment.category">
+        <field name="name">Servers</field>
     </record>
 
+    <!-- <record id="demo_inventory_server" model="maintenance.equipment"> -->
+    <!--     <field name="name">Odoo-Server</field> -->
+    <!--     <field name="category_id" ref="demo_inventory_category_servers"/> -->
+    <!--     <field name="equipment_assign_to">employee</field> -->
```
Comment (janikvonrotz) [inline mgmt_asset/demo/demo.xml]: This should be given as a default.

File: mgmt_audit/demo/demo.xml [hunk 24]
```diff
@@ -1,23 +1,64 @@
...
+    <record id="demo_audit_stage_ready" model="mgmt.audit.stage">
+        <field name="name">Ready</field>
+        <field name="code">ready</field>
+        <field name="sequence">2</field>
     </record>
 
-    <record id="demo_statement0" model="mgmt.statement">
-        <field name="name">Statement 0</field>
+    <record id="demo_audit_stage_in_progress" model="mgmt.audit.stage">
+        <field name="name">In progress</field>
```
Comment (janikvonrotz) [inline mgmt_audit/demo/demo.xml]: "In Progress"

File: mgmt_audit_project/demo/demo.xml [hunk 25]
```diff
@@ -1,15 +1,12 @@
...
-   <record id="demo_task0" model="project.task">
-        <field name="name">Task 0</field>
-    </record>
+  <record id="demo_nonconformity_no_power_supply" model="mgmt.nonconformity">
+    <field name="name">No Redundant Power Supply</field>
+    <field name="type">major</field>
+  </record>
+
+  <record id="demo_task_install_power_supply" model="project.task">
+    <field name="name">Install power supply</field>
```
Comment (janikvonrotz) [inline mgmt_audit_project/demo/demo.xml:9]: The task needs to be linked to the nonconfirmity

**Review by janikvonrotz — CHANGES_REQUESTED** [review]

File: mgmt_asset/demo/demo.xml (hunk 23 above)
Comment (laszlokr) [reply mgmt_asset/demo/demo.xml]: @janikvonrotz sorry, but I don't quite understand how to make this work. The `equipment_assign_to` is a selection field: 
image
I provide a value "employee" to this field, but it is invalid. I can create same record manually in the UI and it does not require assigning to a specific user. It is absolutely not clear for me why this demo data would not work and what needs to be changed. I understand that extending `maintenance.equipment` is not an option either. Please give some advice on how to resolve this as I am already spending way too much time on this one

Comment (janikvonrotz) [reply mgmt_asset/demo/demo.xml]: @laszlokr Did you check the code of maintenance? The field has a default value: https://github.com/odoo/odoo/blob/65d4ff2ed145db978843d7f2f04263956c4c0d96/addons/hr_maintenance/models/equipment.py#L13
Therefore I expect that you don't have to define this field.

What is the actual issue with the demo data? Are you sure you did not get side tracked on this issue?

**Comment by janikvonrotz** [comment]
@laszlokr I am currently fixing the PR.

**Comment by janikvonrotz** [comment]
@laszlokr The form definitions are not what I expected. Some fields should have been added to a separate tab and widgets="many2many_tags" should be avoided.

**Comment by janikvonrotz** [comment]
@laszlokr Moreover in some places you added tabs but forgot to give a name.

File: mgmt_asset/demo/demo.xml (hunk 23 above)
Comment (laszlokr) [reply mgmt_asset/demo/demo.xml]: @janikvonrotz I haven't checked that it has a default value, but it does not change much. Here is the error when the field is not defined:
```
psycopg2.errors.NotNullViolation: null value in column "equipment_assign_to" violates not-null constraint
DETAIL:  Failing row contains (12, 1, null, null, null, null, null, null, 6, null, null, 1, 1, null, null, null, null, 2024-09-18, null, null, null, {"en_US": "Odoo-Server"}, null, t, 2024-09-18 10:01:30.55465, 2024-09-18 10:01:30.55465, null, null, null, null).

    raise ParseError('while parsing %s:%s, somewhere inside\n%s' % (
odoo.tools.convert.ParseError: while parsing /home/laszlo/work/mintsystem/Odoo-Build/addons/management-system/mgmt_asset/demo/demo.xml:7, somewhere inside
<record id="demo_inventory_server" model="maintenance.equipment">
        <field name="name">Odoo-Server</field>
        <field name="category_id" ref="demo_inventory_category_servers"/>
    </record>
```

When the field is defined but selection not specified:
```
    raise ValueError("Invalid field %r on model %r" % (key, self._name))
ValueError: Invalid field 'equipment_assign_to' on model 'maintenance.equipment'

odoo.tools.convert.ParseError: while parsing /home/laszlo/work/mintsystem/Odoo-Build/addons/management-system/mgmt_asset/demo/demo.xml:7, somewhere inside
<record id="demo_inventory_server" model="maintenance.equipment">
        <field name="name">Odoo-Server</field>
        <field name="category_id" ref="demo_inventory_category_servers"/>
        <field name="equipment_assign_to"/>
    </record>
```

When the field is defined and selection is specified, same goes for any other non default selection:
```
    raise ValueError("Invalid field %r on model %r" % (key, self._name))
ValueError: Invalid field 'equipment_assign_to' on model 'maintenance.equipment'

odoo.tools.convert.ParseError: while parsing /home/laszlo/work/mintsystem/Odoo-Build/addons/management-system/mgmt_asset/demo/demo.xml:7, somewhere inside
<record id="demo_inventory_server" model="maintenance.equipment">
        <field name="name">Odoo-Server</field>
        <field name="category_id" ref="demo_inventory_category_servers"/>
        <field name="equipment_assign_to">employee</field>
    </record>
```

Not sure where I could get side tracked, but of course that's also possible. I just tried all options there are and still cannot make this piece of demo data load

**Comment by janikvonrotz** [comment]
@laszlokr Some fields such as `stage` should be added as a widget statusbar.

**Comment by janikvonrotz** [comment]
@laszlokr The `asset_ids` field was added to `mgmt_asset` instead of `mgmt_asset_risk`

**Comment by janikvonrotz** [comment]
@laszlokr You forgot to run `task lint` in the module folder.

**Comment by laszlokr** [comment]
> @laszlokr I am currently fixing the PR.

Thank you very much

**Comment by janikvonrotz** [comment]
@laszlokr I pushed my changes. I did not have any problem with the inventory demo data. It worked as I expected.

**Comment by laszlokr** [comment]
> @laszlokr The `asset_ids` field was added to `mgmt_asset` instead of `mgmt_asset_risk`

This is not specified in ERD

**Comment by laszlokr** [comment]
> @laszlokr You forgot to run `task lint` in the module folder.

I did run it, but not before latest commits, sorry for that

**Comment by laszlokr** [comment]
> @laszlokr I pushed my changes. I did not have any problem with the inventory demo data. It worked as I expected.

@janikvonrotz I am really curious what I was doing wrong, still don't understand why it didn't work

**Comment by janikvonrotz** [comment]
@laszlokr I updated the demo data.

**Comment by janikvonrotz** [comment]
@laszlokr Did you reset the database from time to time?

**Comment by janikvonrotz** [comment]
@laszlokr 

> This is not specified in ERD

The ERD does not tell you to which module each field belongs.

However, when reading the specification the definition of a bridge module in this case `mgmt_asset_risk` implies that a connection between the asset and risk model is done. In this case it is the `asset_ids` field.

**Comment by laszlokr** [comment]
> @laszlokr Did you reset the database from time to time?

@janikvonrotz yes, I did reset it a few times while working on these modules

File: mgmt_asset/demo/demo.xml [hunk 26]
```diff
@@ -1,7 +1,17 @@
...
+    <record id="demo_inventory_category_servers" model="maintenance.equipment.category">
+        <field name="name">Servers</field>
+    </record>
+
+    <record id="demo_inventory_server" model="maintenance.equipment">
+        <field name="name">Odoo-Server</field>
+        <field name="category_id" ref="demo_inventory_category_servers" />
+    </record>
+
+    <record id="demo_asset_server" model="mgmt.asset">
```
Comment (janikvonrotz) [inline mgmt_asset/demo/demo.xml:12]: Setting up the asset might be difficult. Check if you can use `eval="ref('mgmt_asset.demo_inventory_server').id"` to get the record id.

I updated the ERD. The model_name is required and model_id is computed from model_name.

File: mgmt_asset/models/asset.py [hunk 27]
```diff
@@ -9,4 +9,30 @@ class MgmtAsset(models.Model):
...
+    @api.depends("model_name", "model_id", "res_id")
+    def _compute_name(self):
+        for record in self:
+            if record.model_name and record.res_id:
+                try:
+                    linked_record = self.env[record.model_name].browse(record.res_id)
+                    record.name = (
+                        linked_record.display_name if linked_record else "Unknown"
+                    )
+                except KeyError:
```
Comment (janikvonrotz) [inline mgmt_asset/models/asset.py]: Remove the `try` and `except` it is no onger required once the model_name is required.

File: mgmt_audit/views/audit_views.xml [hunk 28]
```diff
@@ -1,12 +1,17 @@
...
+  <record id="audit_form_view" model="ir.ui.view">
         <field name="name">mgmt_audit.audit_form_view</field>
         <field name="model">mgmt.audit</field>
         <field name="arch" type="xml">
             <form>
                 <header>
+                    <field
+                        name="stage_id"
+                        widget="statusbar"
+                        statusbar_visible="draft,in_progress,done"
```
Comment (janikvonrotz) [inline mgmt_audit/views/audit_views.xml]: I think the ready state is missing here.

File: mgmt_audit/views/audit_views.xml [hunk 29]
```diff
@@ -21,23 +26,80 @@
...
+                                    <field name="type" />
+                                    <field name="planned_date" />
+                                    <field name="stage_id" />
+                                    <field name="responsible_id" />
+                                    <field name="auditor_ids" />
                                 </group>
                             </group>
                         </page>
+                        <page string="Requirements" name="requirements">
+                            <field name="requirement_ids" widget="many2many_tags" />
```
Comment (janikvonrotz) [inline mgmt_audit/views/audit_views.xml]: Remove the `widget="many2many_tags"` and create an tree list.

File: mgmt_audit/views/audit_views.xml [hunk 30]
```diff
@@ -21,23 +26,80 @@
...
+                <field name="name" />
+                <field name="type" />
+                <field name="planned_date" />
+                <field name="stage_id" />
+                <field name="responsible_id" />
+
+                <templates>
+                    <t t-name="kanban-box">
+                        <div t-attf-class="oe_kanban_global_click">
+                            <div class="o_kanban_details">
```
Comment (janikvonrotz) [inline mgmt_audit/views/audit_views.xml]: Not sure about this kanban format.

There should be a title element.

File: mgmt_audit/views/nonconformity_views.xml [hunk 31]
```diff
@@ -21,6 +21,7 @@
                             <group>
                                 <group>
                                     <field name="name" />
```
Comment (janikvonrotz) [inline mgmt_audit/views/nonconformity_views.xml]: Why is name in the group here?

File: mgmt_hazard_risk/models/hazard.py
Comment (janikvonrotz) [inline mgmt_hazard_risk/models/hazard.py:1]: There should also be a `risk.py` that overwrites the calculation of the risk score based on the hazards.

File: mgmt_requirement/views/paragraph_views.xml [hunk 32]
```diff
@@ -21,6 +21,9 @@
                             <group>
                                 <group>
                                     <field name="name" />
```
Comment (janikvonrotz) [inline mgmt_requirement/views/paragraph_views.xml]: Remove the name field.

**Review by janikvonrotz — CHANGES_REQUESTED** [review]

File: mgmt_audit_project/views/nonconformity_views.xml [hunk 33]
```diff
@@ -6,8 +6,8 @@
         <field name="model">mgmt.nonconformity</field>
         <field name="inherit_id" ref="mgmt_audit.nonconformity_form_view" />
         <field name="arch" type="xml">
-            <field name="name" position="after">
-                <field name="name" />
+            <field name="type" position="after">
+                <field name="task_ids" />
```
Comment (janikvonrotz) [inline mgmt_audit_project/views/nonconformity_views.xml]: Should be a tree view.

File: mgmt_audit_project/views/project_task_views.xml [hunk 34]
```diff
@@ -12,4 +12,18 @@
         </field>
     </record>
 
+    <record id="action_project_task_view" model="ir.actions.act_window">
```
Comment (janikvonrotz) [inline mgmt_audit_project/views/project_task_views.xml:15]: Add domain to filter tasks with nonconformities or recommendations.

File: mgmt_risk/models/probability.py [hunk 35]
```diff
@@ -9,4 +9,5 @@ class MgmtProbability(models.Model):
     _name = "mgmt.probability"
```
Comment (janikvonrotz) [inline mgmt_risk/models/probability.py:9]: Whenever a probability or a severity is created. Add combination records to `mgmt.risk.combination`.

use the `def create` and `def unlink` method.

File: mgmt_audit/views/recommendation_views.xml [hunk 36]
```diff
@@ -16,15 +16,15 @@
...
-                    <notebook>
-                        <page string="General Information" name="general_info">
-                            <group>
-                                <group>
-                                    <field name="name" />
-                                </group>
-                            </group>
-                        </page>
-                    </notebook>
+                    <!-- <notebook> -->
```
Comment (janikvonrotz) [inline mgmt_audit/views/recommendation_views.xml:19]: Keep notebook and page.

File: mgmt_audit_project/views/nonconformity_views.xml [hunk 37]
```diff
@@ -1,15 +1,18 @@
...
         <field name="name">mgmt_audit_project.nonconformity_form_view</field>
         <field name="model">mgmt.nonconformity</field>
         <field name="inherit_id" ref="mgmt_audit.nonconformity_form_view" />
         <field name="arch" type="xml">
-            <field name="name" position="after">
-                <field name="name" />
+            <field name="type" position="after">
+                <field name="task_ids">
+                    <tree editable="bottom">
+                        <field name="name"/>
```
Comment (janikvonrotz) [inline mgmt_audit_project/views/nonconformity_views.xml]: Also add the `project_id`.

File: mgmt_audit_project/views/nonconformity_views.xml [hunk 38]
```diff
@@ -1,15 +1,18 @@
...
     <record id="nonconformity_form_view" model="ir.ui.view">
         <field name="name">mgmt_audit_project.nonconformity_form_view</field>
         <field name="model">mgmt.nonconformity</field>
         <field name="inherit_id" ref="mgmt_audit.nonconformity_form_view" />
         <field name="arch" type="xml">
-            <field name="name" position="after">
-                <field name="name" />
+            <field name="type" position="after">
+                <field name="task_ids">
+                    <tree editable="bottom">
```
Comment (janikvonrotz) [inline mgmt_audit_project/views/nonconformity_views.xml]: Remove `editable="bottom"`

File: mgmt_audit/views/audit_views.xml [hunk 39]
```diff
@@ -20,10 +26,33 @@
...
-                                    <field name="name" />
+                                    <field name="type" />
+                                    <field name="planned_date" />
+                                    <field name="responsible_id" />
+                                    <field name="auditor_ids" />
                                 </group>
                             </group>
                         </page>
+                        <page string="Requirements" name="requirements">
+                          <field name="requirement_ids">
```
Comment (janikvonrotz) [inline mgmt_audit/views/audit_views.xml]: Add `editable="bottom"`

File: mgmt_requirement/views/document_views.xml
Comment (janikvonrotz) [inline mgmt_requirement/views/document_views.xml:1]: Keep notebook and page.

File: mgmt_risk/models/risk_combination.py [hunk 40]
```diff
@@ -9,4 +9,13 @@ class MgmtRiskCombination(models.Model):
     _name = "mgmt.risk.combination"
```
Comment (janikvonrotz) [inline mgmt_risk/models/risk_combination.py:9]: Add `_order = "name"`

File: mgmt_risk/models/risk_combination.py [hunk 41]
```diff
@@ -9,4 +9,13 @@ class MgmtRiskCombination(models.Model):
...
     _description = "Mgmt Risk Combination"
 
-    name = fields.Char()
+    name = fields.Char(required=True)
+    severity_id = fields.Many2one("mgmt.severity")
+    probability_id = fields.Many2one("mgmt.probability", required=True)
+
+    @api.model
```
Comment (janikvonrotz) [inline mgmt_risk/models/risk_combination.py]: Remove this.

File: mgmt_risk/models/risk_combination.py [hunk 42]
```diff
@@ -9,4 +9,13 @@ class MgmtRiskCombination(models.Model):
     _name = "mgmt.risk.combination"
     _description = "Mgmt Risk Combination"
 
-    name = fields.Char()
+    name = fields.Char(required=True)
+    severity_id = fields.Many2one("mgmt.severity")
```
Comment (janikvonrotz) [inline mgmt_risk/models/risk_combination.py]: Make them readonly and required.

File: mgmt_risk/views/risk_views.xml [hunk 43]
```diff
@@ -36,8 +40,59 @@
...
+                <field name="stage" />
             </tree>
         </field>
     </record>
 
+    <record id="risk_kanban_view" model="ir.ui.view">
+        <field name="name">mgmt_risk.risk_kanban_view</field>
+        <field name="model">mgmt.risk</field>
+        <field name="arch" type="xml">
+                <kanban class="o_kanban_small_column" default_group_by="stage">
```
Comment (janikvonrotz) [inline mgmt_risk/views/risk_views.xml]: Show the available columns by default.

File: mgmt_audit/models/audit.py
Comment (janikvonrotz) [inline mgmt_audit/models/audit.py:1]: @api.model
    def _read_group_stage_ids(self, stages, domain, order):
        # Return all stages to ensure all columns are visible in kanban view
        return self.env["mgmt.audit.stage"].search([])
        
    stage_id = fields.Many2one(
        "mgmt.audit.stage",
        required=True,
        default=lambda self: self.env.ref("mgmt_audit.request_stage_draft").id,
        group_expand="_read_group_stage_ids",
    )

File: mgmt_audit/demo/demo.xml
Comment (janikvonrotz) [inline mgmt_audit/demo/demo.xml:1]: Move stages to data.xml.

**Review by janikvonrotz — CHANGES_REQUESTED** [review]

File: mgmt_risk/models/risk.py [hunk 44]
```diff
@@ -9,4 +9,36 @@ class MgmtRisk(models.Model):
...
+    risk_owner_id = fields.Many2one("res.users", required=True)
+    severity_id = fields.Many2one("mgmt.severity", required=True)
+    hazard_ids = fields.Many2many("mgmt.hazard", required=True)
+    revision_risk_ids = fields.One2many("mgmt.risk", "head_risk_id")
+    head_risk_id = fields.Many2one("mgmt.risk")
+    probability_id = fields.Many2one("mgmt.probability", required=True)
+    risk_score = fields.Float(compute="_compute_risk_score", store=True)
+    color = fields.Integer(compute="_compute_color", store=True)
+    stage = fields.Selection(
+        [("identify", "Identify"), ("evaluate", "Evaluate"), ("mitigate", "Mitigate")],
```
Comment (janikvonrotz) [inline mgmt_risk/models/risk.py]: Define a new model mgmt.risk.stage and define the stages in a data.xml.

**Review by janikvonrotz — CHANGES_REQUESTED** [review]

**Pull request merged**